### DISCUSSION TITLE
Flush the container in between unit tests

### DISCUSSION THREAD

**Discussion by vinvir** [Azure/Azurite#2465] [Q&A]
Hello,

In my integration tests, I am using azurite (docker).
Is there a way to flush / delete everything in between 2 tests without deleting the container and creating a new one ?

This would be very cool performance wise.

Thanks

**Comment by blueww** [comment]
@vinvir 

I don't see any easy way to do that.
If the container count is not too much, delete the containers and recreate them might be the the fastest way.

Another way is: 
1. Start Azurite with "-l [locationFolder]"
2. setup all thing that need be prepared before run the test (like create containers)
3. Stop Azurite and back up the locationFolder to another folder.
4. Start Azurite with "-l [locationFolder]" and run test
5. After test finish, stop Azurite, clear up location Folder, and copy all content from backup folder to the locationFolder
6. Start Azurite with "-l [locationFolder]" and run the next test

**Comment by blueww** [comment]
You can also run Azurite with "--inMemoryPersistence", then everything will store in memory. 
When you stop and restart Azurite, all stored items will be cleaned up.
